airflow/serialization/serde.py

```diff
@@ -302,6 +308,11 @@ def _stringify(classname: str, version: int, value: T | None) -> str:
     return s
 
 
+def _is_pydantic(cls: Any) -> bool:
+    """Return True if the class is a pydantic model."""
+    return hasattr(cls, "validate") and hasattr(cls, "json") and hasattr(cls, "dict")
```

Comment:
isinstance is slower than checking for attributes, it will need to walk the tree of objects to find out whether it is an instance. It is one of the reasons the new serde code is faster than the old serialization code.

Reply:
How about using `isinstance` _after_ the attribute checks? The problem I have with the current check is those attribute names are quite common and can easily hit false positives. If we check the attributes first, the probability of the object passed to `isinstance` is a Pydantic model should be high, and since Pydantic models mostly inherit from BaseModel directly, the talk should end very quickly (only two steps).

Reply:
import timeit

def askisinstance(foo=object()):
    return isinstance(foo, BaseModel)

def askhasattr(foo=object()):
    return hasattr(foo, "validate") and hasattr(foo, "dict") and hasattr(foo, "json")



#%%
timeit.timeit('testfunc()', 'from __main__ import askisinstance as testfunc')
timeit.timeit('testfunc()', 'from __main__ import askhasattr as testfunc')

# instance
0.19626916700508446

# hasattr
0.06582462496589869

Reply:
updated to check for different attributes set by pydantic metaclass.